src/hip/hip_build_utils.cpp

```diff
@@ -107,8 +107,8 @@ const char* getOffloadBundlerBinPath()
 
 namespace miopen {
 
-static fs::path HipBuildImpl(boost::optional<TmpDir>& tmp_dir,
-                             const std::string& filename,
+static fs::path HipBuildImpl(const TmpDir& tmp_dir,
+                             const fs::path& filename,
```

Comment:
This `filename` is actually a text label that denotes that chunk of source code in our internal storage.

It is Ok to use this text label to construct fs::path objects, however.

Reply:
[resolved][I do not care]